expat/CMakeLists.txt

```diff
@@ -67,6 +71,17 @@ if(XML_NS)
 else(XML_NS)
     set(XML_NS 0)
 endif(XML_NS)
+if(XML_UNICODE)
+    set(XML_UNICODE 1)
+else(XML_UNICODE)
+    set(XML_UNICODE 0)
+endif(XML_UNICODE)
+if(XML_UNICODE_WCHAR_T)
+    set(XML_UNICODE_WCHAR_T 1)
+else(XML_UNICODE_WCHAR_T)
+    set(XML_UNICODE_WCHAR_T 0)
+endif(XML_UNICODE_WCHAR_T)

```

Comment:
For XML_UNICODE without XML_UNICODE_WCHAR_T we may need to exclude xmlwf from compilation to not have the user run into compile errors.

Also, XML_UNICODE_WCHAR_T auto-activates/requires XML_UNICODE in the inside of Expat. I wonder if that nature should be reflected on CMake level somewhere as well.

Reply:
Wouldn't it be better that the user excludes xmlwf using BUILD_tools OFF? Imho silently excluding xmlwf without user request is worse than letting the user see the compile message "xmlwf requires a 16-bit Unicode-compatible wchar_t".

Reply:
It wouldn't be strictly necessary to reflect that fact that XML_UNICODE_WCHAR_T auto-activates/requires XML_UNICODE. We could make it an error if the user wants XML_UNICODE_WCHAR_T but not XML_UNICODE. However, I have no idea how to do this.

Reply:
> Wouldn't it be better that the user excludes xmlwf using BUILD_tools OFF? Imho silently excluding xmlwf without user request is worse than letting the user see the compile message "xmlwf requires a 16-bit Unicode-compatible wchar_t".

Agreed, let's detect it and show an error. Soemthing like `message(SEND_ERROR "... BUILD_tools=OFF ........")` may work (https://cmake.org/cmake/help/v3.0/command/message.html)

> We could make it an error if the user wants XML_UNICODE_WCHAR_T but not XML_UNICODE. However, I have no idea how to do this.

`if`, `AND`, `NOT` and `message` combined should work (https://cmake.org/cmake/help/v3.0/command/if.html?highlight=#command:if)

Reply:
Please review my changes.